src/odemis/util/concurrent.py

```
@@ -0,0 +1,430 @@
+        e, tb = sys.exc_info()[1:]
+        try:
+            future.set_exception_info(e, tb)
+        except AttributeError:  # Old futures (<v3) only had the non-traceback version
+            future.set_exception(e)
+    else:
+        future.set_result(result)
+
+
+# inspect.getmembers() in Python 3.10 and prior has an issue when __getattr__ is modified.
+if sys.version_info >= (3, 11):
+    def inspect_getmembers(object, predicate=None):
```

Ruff (0.15.14)

[error] Function argument `object` is shadowing a Python builtin

(A002)